addons/mail/static/src/owl/components/chatter_topbar/chatter_topbar.js

```diff
@@ -28,15 +30,62 @@ class ChatterTopbar extends Component {
 
     /**
      * @private
-     * @param {Event} ev
+     * @param {MouseEvent} ev
      */
     _onClickAttachments(ev) {
         this.trigger('o-chatter-topbar-select-attachment');
     }
+
+    /**
+     * @private
+     * @param {MouseEvent} ev
+     */
+    _onClickFollow(ev) {
+        this.trigger('o-chatter-topbar-follow');
+    }
+
+    /**
+     * @private
+     * @param {MouseEvent} ev
+     */
+    _onClickFollowers(ev) {
+        this.trigger('o-chatter-topbar-show-followers');
+    }
+
+    /**
+     * @private
+     * @param {MouseEvent} ev
+     */
+    _onClickLogNote(ev) {
+        this.trigger('o-chatter-topbar-log-note');
+    }
+
+    /**
+     * @private
+     * @param {MouseEvent} ev
+     */
+    _onClickScheduleActivity(ev) {
+        this.trigger('o-chatter-topbar-schedule-activity');
+    }
+
+    /**
+     * @private
+     * @param {MouseEvent} ev
+     */
+    _onClickSendMessage(ev) {
+        this.trigger('o-chatter-topbar-send-message');
+    }
 }
 
+ChatterTopbar.defaultProps = {
+    isDisabled: false,
+};
+
 ChatterTopbar.props = {
-    threadLocalId: String,
+    isComposerLog: Boolean,
+    isComposerVisible: Boolean,
+    isDisabled: Boolean,
```

Comment:
I feel like code would be easier to understand if all goes into chatter entity.
That way, less indirections (no local `state` and pass some as props + lots of `trigger()`/`t-on-`), simply `storeProps.chatter` and `storeDispatch()` instead.

Reply:
I don't understand :(